equinox/serialisation.py

```diff
@@ -194,17 +194,16 @@ def tree_deserialise_leaves(
     **Arguments:**
 
     - `path`: The file location to load values from.
-    - `like`: A PyTree of the same structure, and with leaves of the same type, as the
-        PyTree being loaded. Those leaves which are loaded will replace the
-        corresponding leaves of `like`.
+    - `like`: A PyTree of same, prefix or compatible-partial structure
+        of the saved PyTree. The matching leaves are required to be of the same type.
+        Those leaves which are loaded will replace the corresponding leaves of `like`.
```

Comment:
I seem to recall from #136 that we *can't* deserialise prefixes at all. Indeed we iterate over `like` in a depth-first order, which I would have thought means we end up deserialising leaves in the wrong order.

I note that the test and example you've written for this isn't a PyTree prefix -- it's a PyTree of the same structure (an `eqx.nn.MLP` with leaves of a different shape (the weight/bias for the final layer).

Reply:
Yes, my bad. I'll try with the Prefix but I think it won't work, like you said. 
For the MLP I wanted to check the `DFS` matching substructure bit.